charts/redpanda/post_install_upgrade_job.go

```diff
@@ -0,0 +1,200 @@
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=post-install-upgrade-job.go.tpl
+package redpanda
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+)
+
+func PostInstallUpgradeJob(dot *helmette.Dot) *batchv1.Job {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if !values.PostInstallJob.Enabled {
+		return nil
+	}
+
+	job := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-configuration", Fullname(dot)),
+			Namespace: dot.Release.Namespace,
+			Labels: helmette.Merge(
+				FullLabels(dot),
+				helmette.Default(map[string]string{}, values.PostInstallJob.Labels),
+			),
+			Annotations: helmette.Merge(
+				// This is what defines this resource as a hook. Without this line, the
+				// job is considered part of the release.
+				map[string]string{
+					"helm.sh/hook":               "post-install,post-upgrade",
+					"helm.sh/hook-delete-policy": "before-hook-creation",
+					"helm.sh/hook-weight":        "-5",
+				},
+				helmette.Default(map[string]string{}, values.PostInstallJob.Annotations),
+			),
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: fmt.Sprintf("%s-post-", dot.Release.Name),
+					Labels: helmette.Merge(
+						map[string]string{
+							"app.kubernetes.io/name":      Name(dot),
+							"app.kubernetes.io/instance":  dot.Release.Name,
+							"app.kubernetes.io/component": fmt.Sprintf("%s-post-install", Name(dot)[:50]),
+						},
+						helmette.Default(map[string]string{}, values.CommonLabels),
+					),
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:     values.NodeSelector,
+					Affinity:         postInstallJobAffinity(dot),
+					Tolerations:      tolerations(dot),
+					RestartPolicy:    corev1.RestartPolicyNever,
+					SecurityContext:  PodSecurityContext(dot),
+					ImagePullSecrets: pullSecrets(dot),
+					Containers: []corev1.Container{
+						{
+							Name:      fmt.Sprintf("%s-post-install", Name(dot)),
+							Image:     fmt.Sprintf("%s:%s", values.Image.Repository, Tag(dot)),
+							Env:       PostInstallUpgradeEnvironmentVariables(dot),
+							Command:   []string{"bash", "-c"},
+							Args:      []string{},
+							Resources: postInstallJobResources(values.PostInstallJob.Resources),
```

Comment:
I just wonder why not:

```suggestion
							Resources: helmette.MergeTo[corev1.ResourceRequirements](resources),
```

Reply:
Yeah, given how trivial that function became once I had MergeTo I'll do this.

Reply:
... actually there's something amiss here. That refactor leaves the resulting template failing when values.postInstallJob.resources is empty.